modules/kafka/src/main/scala/ch/epfl/bluebrain/nexus/service/kafka/KafkaSupervisedConsumer.scala

```diff
@@ -0,0 +1,140 @@
+                                     index: Event => Future[Unit],
+                                     topic: String,
+                                     decoder: Decoder[Event],
+                                     committable: Boolean,
+                                     failuresLog: Option[IndexFailuresLog])
+    extends Actor {
+
+  private implicit val as: ActorSystem       = context.system
+  private implicit val ec: ExecutionContext  = as.dispatcher
+  private implicit val mt: ActorMaterializer = ActorMaterializer()
+
+  private val log     = Logger[this.type]
+  private val config  = context.system.settings.config.getConfig("indexing.retry")
+  private val retries = config.getInt("max-count")
+  private val strategy =
+    Backoff(Duration(config.getDuration("max-duration", SECONDS), SECONDS), config.getDouble("random-factor"))
+
+  override val receive: Receive = emptyBehavior
+
+  override def preStart(): Unit = {
+    super.preStart()
+    val done = if (committable) {
+      Consumer
+        .committableSource(settings, Subscriptions.topics(topic))
+        .mapAsync(1) { msg =>
+          process(msg)
+            .recover { case e => log.error(s"Unexpected failure while processing message: $msg", e) }
+            .map(_ => msg.committableOffset)
+        }
+        .batch(max = 20, first => CommittableOffsetBatch.empty.updated(first)) { (batch, elem) =>
+          batch.updated(elem)
+        }
+        .mapAsync(1)(_.commitScaladsl())
+        .runWith(Sink.ignore)
+    } else {
+      Consumer
+        .plainSource(settings, Subscriptions.topics(topic))
+        .mapAsync(1) { record =>
+          process(record)
+            .recover { case e => log.error(s"Unexpected failure while processing record: $record", e) }
+        }
+        .runWith(Sink.ignore)
+    }
+
+    done.onComplete {
+      case Failure(e) =>
+        log.error("Stream failed with an error, stopping the actor and restarting.", e)
+        kill()
+      // $COVERAGE-OFF$
+      case Success(_) =>
+        log.warn("Kafka consumer stream ended gracefully, however this should not happen; restarting.")
+        kill()
+      // $COVERAGE-ON$
+    }
+  }
+
+  private def kill(): Unit = self ! PoisonPill
+
+  private def process(record: ConsumerRecord[String, String]): Future[Unit] = {
+    parse(record.value).flatMap(decoder.decodeJson) match {
+      case Right(event) =>
+        log.debug(s"Received message: ${record.value}")
+        (() => index(event))
+          .retry(retries)(strategy)
+          .recoverWith {
+            // $COVERAGE-OFF$
+            case e =>
+              log.error(s"Failed to index event: $event; skipping.", e)
+              storeFailure(record)
+          }
+      case Left(e) =>
+        log.error(s"Failed to decode message: ${record.value}; skipping.", e)
+        storeFailure(record)
+      // $COVERAGE-ON$
+    }
+  }
+
+  private def process(msg: CommittableMessage[String, String]): Future[Unit] = {
```

Comment:
call the overloaded method with an optional offset argument that defaults to `None`